Inseok_45/app.js

```diff
@@ -94,16 +101,94 @@ app.get('/users/:userId/post/view', async (req, res, next) => {
           ) AS postings
         FROM users
         WHERE users.id = ?
-        `, [userId, userId]
+        `, [ userId, userId ]
     );
 
     return res.status(200).json({ data: userPost }); 
 });
 
+app.patch('/post', async (req, res, next) => {
+
+    const { userId, postId, postContent } = req.body;
+    
+    await dataSource.query(
+        `UPDATE
+          posts
+        SET post_paragraph = ${ postContent }
+        WHERE posts.user_id = ? && posts.id = ?
+        `, [ userId, postId ]
+    );
+
+    const [editPost] = await dataSource.query(
+        `SELECT
+          posts.user_id AS userId,
+          users.user_name,
+          posts.id AS postingId,
+          posts.post_image AS postImage,
+          posts.post_paragraph AS postContent
+        FROM
+          posts
+        INNER JOIN users
+        WHERE posts.user_id = ? && posts.id = ?
+        `, [ userId, postId ]
+    );
+
+    return res.status(201).json({ data : "Post Update Successful" });
+});
+
 
 
+app.put('/like', async (req, res, next) => {    
 
+    const { userId, postId } = req.body;
+    
+    const newLike = await dataSource.query(
+        `SELECT
+        EXISTS
+        (SELECT * 
+        FROM likes
+        WHERE user_id = ? AND post_id = ?)
+        `, [ userId, postId ]
+    );
+    console.log(newLike);
+    const rArr = Object.values(newLike[0]);
+    const checkExist = Number(rArr[0]);
+    
+    if(checkExist === 0) {
+        await dataSource.query(
+            `INSERT INTO likes(
+                user_id,
+                post_id
+            ) VALUE ( ?, ?)
+            `, [ userId, postId ]
+        );
+        return res.status(201).json({ message : " Like Successful" });
+
+    }
+        await dataSource.query(
+            `DELETE FROM
+            likes
+            WHERE
+            user_id = ? AND post_id = ?
+            `, [ userId, postId ]
+        );
+        return res.status(200).json({ message : " Unlike Successful" });
 
+});
+
+app.delete('/users/posts/del', async (req, res, next) => {      
```

Comment:
```suggestion
app.delete('/users/posts', async (req, res, next) => {      
```
아예 안들어가도 http request method로 동작 의미 설명 충분합니다..ㅎㅎ